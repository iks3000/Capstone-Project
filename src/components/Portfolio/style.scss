@import "../../assets/styles/_variables.scss";
@import "../../assets/styles/_extends.scss";
.portfolio {
    &__title {
        @extend %title;
    }
}

.isotope-content {
    &__filter {
        display: flex;
        justify-content: space-between;
        width: 105px;
        cursor: default;
        font: 400 16px/19px $primary-fonts;
        margin-bottom: 20px;
        span {
            cursor: pointer;
        }
        .is-active {
            color: $green;
        }
        .in-active {
            color: $black;
        }
    }
    .filter-item {
        display: block;
        position: relative;
        width: 308px;
        height: 190px;
        margin: 0 20px 20px 0;
        overflow: hidden;
        cursor: default;
        &:hover {
            .info-block {
                transform: translate(0, 0);
            }
        }
    }
    .info-block {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        position: absolute;
        transform: translate(0, 190px);
        padding: 20px 15px;
        width: 100%;
        height: 100%;
        background: hsla(0, 0%, 100%, 0.9);
        transition: transform 0.3s ease-in-out;
        h3 {
            @extend %content-title;
            color: $green;
        }
        a {
            font: 400 14px/16px $primary-fonts;
            color: $green;
            text-decoration: underline;
            width: 100px;
        }
    }
    .code {
        background: url("../../assets/images/card-code.png") 0 0 no-repeat;
        background-size: cover;
    }
    .ui {
        background: url("../../assets/images/card-ui.png") 0 0 no-repeat;
        background-size: cover;
    }
}

@media only screen and (max-width: 600px) {
    .isotope-content {
        .filter-item {
            width: 100%;
            height: 150px;
            margin: 10px 0;
        }
        .info-block {
            padding: 5px 10px;
        }
    }
}