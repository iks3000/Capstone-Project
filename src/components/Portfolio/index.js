/* eslint-disable jsx-a11y/anchor-is-valid */
import { useState, useRef, useEffect } from "react";
import Info from "../../components/Info";
import Isotope from "isotope-layout";
import "./style.scss";

export const IsotopeContent = () => {
  const isotope = useRef();
  const [filterKey, setFilterKey] = useState("*");

  useEffect(() => {
    isotope.current = new Isotope(".filter-container", {
      itemSelector: ".filter-item",
      layoutMode: "fitRows",
    });
    return () => isotope.current.destroy();
  }, []);

  useEffect(() => {
    filterKey === "*"
      ? isotope.current.arrange({ filter: `*` })
      : isotope.current.arrange({ filter: `.${filterKey}` });
  }, [filterKey]);

  const handleFilterKeyChange = (key) => () => {
    setFilterKey(key);
  };

  return (
    <div className="isotope-content">
      <div className="isotope-content__filter">
        <span className={filterKey === "*" ? "is-active" : "in-active"}
          onClick={handleFilterKeyChange("*")}>All</span>
        /
        <span className={filterKey === "code" ? "is-active" : "in-active"}
          onClick={handleFilterKeyChange("code")}>Code</span>
        /
        <span className={filterKey === "ui" ? "is-active" : "in-active"}
          onClick={handleFilterKeyChange("ui")}>UI</span>
      </div>

      <ul className="filter-container">
        <div className="filter-item code">
          <div className="info-block">
            <h3>Some text</h3>
            <Info
              text={
                "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis"
              }
            />
            <a href="#">View resource</a>
          </div>
        </div>
        <div className="filter-item ui">
          <div className="info-block">
            <h3>Some text</h3>
            <Info text={"Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis"}/>
            <a href="#">View resource</a>
          </div>
        </div>
        <div className="filter-item code">
          <div className="info-block">
            <h3>Some text</h3>
            <Info text={"Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis"}/>
            <a href="#">View resource</a>
          </div>
        </div>
        <div className="filter-item ui">
          <div className="info-block">
            <h3>Some text</h3>
            <Info text={"Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis"}/>
            <a href="#">View resource</a>
          </div>
        </div>
      </ul>
    </div>
  );
};

const Portfolio = () => {
  return (
    <div className="portfolio" id="portfolioComponent">
      <h2 className="portfolio__title">Portfolio</h2>
      <IsotopeContent />
    </div>
  );
};

export default Portfolio;
