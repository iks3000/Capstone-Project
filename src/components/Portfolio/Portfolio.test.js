import React from "react";
import { shallow } from "enzyme";
import Portfolio, { IsotopeContent } from "./index";

it("should render Portfolio component", () => {
  expect(shallow(<Portfolio />)).toMatchSnapshot();
});

it("should render IsotopeContent component", () => {
  expect(shallow(<IsotopeContent />)).toMatchSnapshot();
});

// describe("Filter key", () => {
//   const key = "*";
//   const props = {
//     handleFilterKeyChange: jest.fn(),
//     key,
//   };

//   it("should", () => {
//     const button = shallow(
//       <IsotopeContent {...props}>Increment</IsotopeContent>
//     );
//     button.find(".increment").simulate("click");
//     expect(props.handleFilterKeyChange).toHaveBeenCalledWith("ui");
//   });
// });
