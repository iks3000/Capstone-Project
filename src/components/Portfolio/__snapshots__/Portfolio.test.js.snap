// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Portfolio component 1`] = `
<div
  className="portfolio"
  id="portfolioComponent"
>
  <h2
    className="portfolio__title"
  >
    Portfolio
  </h2>
  <IsotopeContent />
</div>
`;
